### ISSUE TITLE
`reportImportCycles` false positives

### ISSUE THREAD

**Issue by lucascolley** [microsoft/pyright#9510]
**Describe the bug**
I am seeing false positives of `reportImportCycles` on a branch of my repo. I can import functions from `_funcs.py` via the module `__init__.py` without error.

```
array-api-extra/src/array_api_extra/__init__.py
  array-api-extra/src/array_api_extra/__init__.py: error: Cycle detected in import chain
    array-api-extra/src/array_api_extra/__init__.py
    array-api-extra/src/array_api_extra/_funcs.py (reportImportCycles)
  array-api-extra/src/array_api_extra/__init__.py: error: Cycle detected in import chain
    array-api-extra/src/array_api_extra/__init__.py
    array-api-extra/src/array_api_extra/_funcs.py
    array-api-extra/src/array_api_extra/_utils.py (reportImportCycles)
2 errors, 0 warnings, 0 informations 
```

**Code or Screenshots**
https://github.com/lucascolley/array-api-extra/tree/isin

**VS Code extension or command-line**
CLI, `pyright 1.1.389`

**Comment by erictraut** [comment]
This isn't a false positive. Pyright's import cycle detection is more strict than the runtime. It is intended to detect architectural layering violations. 

The `reportImportCycles` check is off by default. If you don't want to use it, you can leave it disabled.

**Comment by erictraut** [comment]
Pyright reports import cycles from a static analysis standpoint. For example, if module A imports module B which imports module C which imports module A. The import cycle is indicated in the error message.

From the error message that you've posted above, it appears that module `array-api-extra/src/array_api_extra/__init__.py` is importing module `array-api-extra/src/array_api_extra/_funcs.py` which is in turn importing the previous module.

Note that some imports are implicit. For example, if you import `a.b`, you are implicitly importing both `a` and `a.b`.

**Comment by lucascolley** [comment]
Ah - the culprit must be [this line](https://github.com/lucascolley/array-api-extra/blob/bcae393b0b4e986f37f4cae6d1eede030e06b573/src/array_api_extra/_funcs.py#L9) in which I do a relative `from . import _utils`.

Is there a way to perform that import without triggering this error? Am I supposed to nest `_utils` in a directory if I want to import it 'sideways' as a module?
